spec/vsphere_connection_handling_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'spec_helper'
+require 'chef/knife/base_vsphere_command'
+
+describe Chef::Knife::BaseVsphereCommand do
+  describe '#password' do
+    before do
+      expect(subject).to receive(:get_config).with(:vsphere_pass).at_least(:once).and_return(password)
+    end
+
+    context 'password is in config file' do
+      let(:password) { 'ossifrage' }
+
+      it 'returns the password' do
+        expect(subject.password).to eq 'ossifrage'
+      end
+    end
+
+    context 'password is in config file but encoded' do
+      let(:password) { 'base64:c3F1ZWVtaXNo' }
+
+      it 'decodes the password' do
+        expect(subject.password).to eq 'squeemish'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.